Core/Graphics/Source/FrameBufferManager.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "FrameBuffer.h"
+#include <list>
+#include <memory>
+#include <mutex>
+
+namespace Babylon
+{
+    class FrameBufferManager final
+    {
+    public:
+        FrameBufferManager();
+        FrameBufferManager(const FrameBufferManager&) = delete;
+        FrameBufferManager(FrameBufferManager&&) = delete;
+
+        FrameBuffer& DefaultFrameBuffer();
+
+        FrameBuffer& AddFrameBuffer(bgfx::FrameBufferHandle handle, uint16_t width, uint16_t height, bool backBuffer);
+        void RemoveFrameBuffer(const FrameBuffer& frameBuffer);
+
+        bgfx::ViewId NewViewId();
+        void Reset();
+
+    private:
+        void ResetViewId();
+        void ResetFrameBuffers();
+
+        std::mutex m_viewIdMutex{};
+        bgfx::ViewId m_nextViewId{};
+
+        std::mutex m_frameBuffersMutex{};
+        std::list<std::unique_ptr<FrameBuffer>> m_frameBuffers{};
+
+        FrameBuffer m_default;
```

Comment:
This is intentionally not initialized here since it is initialized by the constructor, right?

Reply:
It doesn't have a default constructor, so it has to be initialized by the manager's constructor.